operator/pkg/ingress/service.go

```diff
@@ -13,6 +14,7 @@ import (
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 
+	operatorOption "github.com/cilium/cilium/operator/option"
```

Comment:
we tried to avoid using global flag directly, as the flags might not be loaded, and caused tricky bug. Currently, new unit test is failing for the same reason.

Alternative, you can take a look at this patch

```diff
diff --git a/operator/cmd/root.go b/operator/cmd/root.go
index d1b67331f0..df25ae5a92 100644
--- a/operator/cmd/root.go
+++ b/operator/cmd/root.go
@@ -595,7 +595,8 @@ func OnOperatorStartLeading(ctx context.Context) {
 		ingressController, err := ingress.NewIngressController(
 			ingress.WithHTTPSEnforced(operatorOption.Config.EnforceIngressHTTPS),
 			ingress.WithSecretsSyncEnabled(operatorOption.Config.EnableIngressSecretsSync),
-			ingress.WithSecretsNamespace(operatorOption.Config.IngressSecretsNamespace))
+			ingress.WithSecretsNamespace(operatorOption.Config.IngressSecretsNamespace),
+			ingress.WithLBAnnotations(operatorOption.Config.IngressLBAnnotations))
 		if err != nil {
 			log.WithError(err).WithField(logfields.LogSubsys, ingress.Subsys).Fatal(
 				"Failed to start ingress controller")
diff --git a/operator/pkg/ingress/ingress.go b/operator/pkg/ingress/ingress.go
index f5739aec61..180869d76d 100644
--- a/operator/pkg/ingress/ingress.go
+++ b/operator/pkg/ingress/ingress.go
@@ -72,6 +72,7 @@ type IngressController struct {
 	enforcedHTTPS      bool
 	enabledSecretsSync bool
 	secretsNamespace   string
+	lbAnnotations      []string
 }
 
 // NewIngressController returns a controller for ingress objects having ingressClassName as cilium
@@ -89,6 +90,7 @@ func NewIngressController(options ...Option) (*IngressController, error) {
 		enforcedHTTPS:      opts.EnforcedHTTPS,
 		enabledSecretsSync: opts.EnabledSecretsSync,
 		secretsNamespace:   opts.SecretsNamespace,
+		lbAnnotations:      opts.LBAnnotations,
 	}
 	ic.ingressStore, ic.ingressInformer = informer.NewInformer(
 		utils.ListerWatcherFromTyped[*slim_networkingv1.IngressList](k8s.WatcherClient().NetworkingV1().Ingresses(corev1.NamespaceAll)),
@@ -327,7 +329,7 @@ func getIngressKeyForService(service *slim_corev1.Service) string {
 }
 
 func (ic *IngressController) createLoadBalancer(ingress *slim_networkingv1.Ingress) error {
-	svc := getServiceForIngress(ingress)
+	svc := getServiceForIngress(ingress, ic.lbAnnotations)
 	svcKey, err := cache.MetaNamespaceKeyFunc(svc)
 	if err != nil {
 		log.Warn("Failed to get service key for ingress")
diff --git a/operator/pkg/ingress/ingress_options.go b/operator/pkg/ingress/ingress_options.go
index 66ef92a7d0..9257a11719 100644
--- a/operator/pkg/ingress/ingress_options.go
+++ b/operator/pkg/ingress/ingress_options.go
@@ -9,6 +9,7 @@ type Options struct {
 	EnforcedHTTPS      bool
 	EnabledSecretsSync bool
 	SecretsNamespace   string
+	LBAnnotations      []string
 }
 
 // DefaultIngressOptions specifies default values for cilium ingress controller.
@@ -16,6 +17,7 @@ var DefaultIngressOptions = Options{
 	MaxRetries:         10,
 	EnforcedHTTPS:      true,
 	EnabledSecretsSync: true,
+	LBAnnotations:      []string{},
 }
 
 // Option customizes the configuration of cilium ingress controller
@@ -52,3 +54,11 @@ func WithSecretsNamespace(secretsNamespace string) Option {
 		return nil
 	}
 }
+
+// WithLBAnnotations configures LB annotations to be used for LB service
+func WithLBAnnotations(lbAnnotations []string) Option {
+	return func(o *Options) error {
+		o.LBAnnotations = lbAnnotations
+		return nil
+	}
+}
diff --git a/operator/pkg/ingress/service.go b/operator/pkg/ingress/service.go
index cf3549acc1..d7c5ca7628 100644
--- a/operator/pkg/ingress/service.go
+++ b/operator/pkg/ingress/service.go
@@ -14,7 +14,6 @@ import (
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 
-	operatorOption "github.com/cilium/cilium/operator/option"
 	"github.com/cilium/cilium/pkg/k8s"
 	"github.com/cilium/cilium/pkg/k8s/informer"
 	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
@@ -187,10 +186,10 @@ func (sm *serviceManager) notify(service *slim_corev1.Service) {
 	}
 }
 
-func getServiceForIngress(ingress *slim_networkingv1.Ingress) *v1.Service {
+func getServiceForIngress(ingress *slim_networkingv1.Ingress, lbAnnotations []string) *v1.Service {
 	annotations := make(map[string]string)
 	for annotationKey, annotationValue := range ingress.ObjectMeta.Annotations {
-		for _, annotationPrefix := range operatorOption.Config.IngressLBAnnotations {
+		for _, annotationPrefix := range lbAnnotations {
 			if strings.HasPrefix(annotationKey, annotationPrefix) {
 				annotations[annotationKey] = annotationValue
 			}
diff --git a/operator/pkg/ingress/service_test.go b/operator/pkg/ingress/service_test.go
index 3f843422da..1c20916105 100644
--- a/operator/pkg/ingress/service_test.go
+++ b/operator/pkg/ingress/service_test.go
@@ -136,7 +136,7 @@ func Test_getServiceForIngress(t *testing.T) {
 		ingress := baseIngress.DeepCopy()
 		ingress.Spec.TLS = nil
 
-		res := getServiceForIngress(ingress)
+		res := getServiceForIngress(ingress, []string{"service.beta.kubernetes.io", "service.kubernetes.io", "cloud.google.com"})
 		assert.Equal(t, "cilium-ingress-dummy-ingress", res.Name)
 		assert.Equal(t, "dummy-namespace", res.Namespace)
 		assert.Equal(t, []metav1.OwnerReference{
@@ -173,7 +173,7 @@ func Test_getServiceForIngress(t *testing.T) {
 			},
 		}
 
-		res := getServiceForIngress(ingress)
+		res := getServiceForIngress(ingress, []string{"service.beta.kubernetes.io", "service.kubernetes.io", "cloud.google.com"})
 		assert.Equal(t, "cilium-ingress-dummy-ingress", res.Name)
 		assert.Equal(t, "dummy-namespace", res.Namespace)
 		assert.Equal(t, []metav1.OwnerReference{

```